src/commands/dockerInstallerBase.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import ChildProcessProvider from '../debugging/coreclr/ChildProcessProvider';
+import { LocalFileSystemProvider } from '../debugging/coreclr/fsProvider';
+import { OSTempFileProvider } from '../debugging/coreclr/tempFileProvider';
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { streamToFile } from '../utils/httpRequest';
+import LocalOSProvider from '../utils/LocalOSProvider';
+
+export abstract class DockerInstallerBase {
+    protected abstract downloadUrl: string;
+    protected abstract fileExtension: string;
+    protected abstract installationMessage: string;
+    protected abstract getInstallCommand(fileName: string): string;
+
+    public async downloadAndInstallDocker(): Promise<void> {
+        const confirmInstall: string = localize('vscode-docker.commands.DockerInstallerBase.confirm', 'Are you sure you want to install docker on this machine.');
+        const installTitle: string = localize('vscode-docker.commands.DockerInstallerBase.install', 'Install');
+        const downloadingMessage: string = localize('vscode-docker.commands.DockerInstallerBase.downloading', 'Downloading Docker installer...');
+        const downloadCompleteMessage: string = localize('vscode-docker.commands.DockerInstallerBase.downloadCompleteMessage', 'Download completed');
+        let downlaodedFileName: string;
+
+        // no need to check result - cancel will throw a UserCancelledError
+        await ext.ui.showWarningMessage(confirmInstall, { modal: true }, { title: installTitle });
```

Comment:
Make sure @BigMorty is OK with a modal dialog, they are very annoying.

Reply:
Here the modal dialog make sense as we need user input to proceed. @BigMorty reopen this comment, if you don't agree.